boa_engine/src/builtins/intl/mod.rs

```diff
@@ -442,9 +385,77 @@ fn insert_unicode_extension_and_canonicalize(locale: &str, extension: &str) -> J
         }
     };
 
-    // TODO 7. Assert: ! IsStructurallyValidLanguageTag(locale) is true.
+    // 7. Assert: ! IsStructurallyValidLanguageTag(locale) is true.
+    let mut new_locale = new_locale
+        .parse()
+        .expect("Assert: ! IsStructurallyValidLanguageTag(locale) is true.");
+
     // 8. Return ! CanonicalizeUnicodeLocaleId(locale).
-    Intl::canonicalize_locale(&new_locale)
+    canonicalize_unicode_locale_id(&mut new_locale, canonicalizer);
+    new_locale.to_string().into()
+}
+
+fn canonicalize_locale_list(args: &[JsValue], context: &mut Context) -> JsResult<Vec<Locale>> {
+    // https://tc39.es/ecma402/#sec-canonicalizelocalelist
```

Comment:
I guess we could add the link to the function documentation, right? And add an explanation of the usage :)